File: src/main/java/br/com/rfasioli/micrometermetrics/messaging/stream/PersonConsumer.java
package br.com.rfasioli.micrometermetrics.messaging.stream;

import br.com.rfasioli.micrometermetrics.persistence.repository.PersonRepository;
import br.com.rfasioli.micrometermetrics.resource.dto.PersonDTO;
import br.com.rfasioli.micrometermetrics.resource.mapper.PersonMapper;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Flux;

import java.util.function.Consumer;

@Component
@RequiredArgsConstructor
public class PersonConsumer implements Consumer<Flux<PersonDTO>> {
  private final PersonMapper personMapper;
  private final PersonRepository personRepository;

  @Override
  public void accept(Flux<PersonDTO> personDTOFlux) {
    personDTOFlux.log()
        .map(personMapper::fromPersonDTO)
        .doOnNext(personRepository::save);
  }
}

File: src/main/kotlin/br/com/rfasioli/micrometermetrics/messaging/stream/PersonConsumer.kt
package br.com.rfasioli.micrometermetrics.messaging.stream

import br.com.rfasioli.micrometermetrics.persistence.document.Person
import br.com.rfasioli.micrometermetrics.persistence.repository.PersonRepository
import br.com.rfasioli.micrometermetrics.resource.dto.PersonDTO
import br.com.rfasioli.micrometermetrics.resource.mapper.toPerson
import org.springframework.stereotype.Component
import reactor.core.publisher.Flux
import java.util.function.Consumer

@Component
class PersonConsumer(
    private val personRepository: PersonRepository
) : Consumer<Flux<PersonDTO>> {
    override fun accept(personDTOFlux: Flux<PersonDTO>) {
        personDTOFlux.log()
            .map(PersonDTO::toPerson)
            .doOnNext(personRepository::save)
    }
}
